src/server/endpoints/covidcast.py

```diff
@@ -111,50 +123,115 @@ def _handle_lag_issues_as_of(q: QueryBuilder, issues: Optional[List[Union[Tuple[
     time_pairs = parse_time_pairs()
     geo_pairs = parse_geo_pairs()
+    jit_bypass = parse_jit_bypass()
 
     as_of = extract_date("as_of")
     issues = extract_dates("issues")
     lag = extract_integer("lag")
+    is_time_type_week = any(time_pair.time_type == "week" for time_pair in time_pairs)
+    is_time_value_true = any(isinstance(time_pair.time_values, bool) for time_pair in time_pairs)
+
+    is_compatibility = is_compatibility_mode()
+    def alias_row(row):
+        if is_compatibility:
+            # old api returned fewer fields
+            remove_fields = ["geo_type", "source", "time_type"]
+            for field in remove_fields:
+                if field in row:
+                    del row[field]
+        if is_compatibility or not alias_mapper or "source" not in row:
+            return row
+        row["source"] = alias_mapper(row["source"], row["signal"])
+        return row
 
     # build query
     q = QueryBuilder(latest_table, "t")
 
-    fields_string = ["geo_value", "signal"]
+    fields_string = ["geo_type", "geo_value", "source", "signal", "time_type"]
     fields_int = ["time_value", "direction", "issue", "lag", "missing_value", "missing_stderr", "missing_sample_size"]
     fields_float = ["value", "stderr", "sample_size"]
-    is_compatibility = is_compatibility_mode()
-    if is_compatibility:
-        q.set_order("signal", "time_value", "geo_value", "issue")
+
+    # TODO: JIT computations don't support time_value = *; there may be a clever way to implement this.
+    use_server_side_compute = not any((issues, lag, is_time_type_week, is_time_value_true)) and JIT_COMPUTE and not jit_bypass
+    if use_server_side_compute:
+        transform_args = parse_transform_args()
+        pad_length = get_pad_length(source_signal_pairs, transform_args.get("smoother_window_length"))
+        time_pairs = pad_time_pairs(time_pairs, pad_length)
+        source_signal_pairs, row_transform_generator = get_basename_signal_and_jit_generator(source_signal_pairs, transform_args=transform_args)
+
+        def gen_transform(rows):
+            parsed_rows = (parse_row(row, fields_string, fields_int, fields_float) for row in rows)
+            transformed_rows = row_transform_generator(parsed_rows=parsed_rows, time_pairs=time_pairs, transform_args=transform_args)
+            for row in transformed_rows:
+                yield alias_row(row)
     else:
-        # transfer also the new detail columns
-        fields_string.extend(["source", "geo_type", "time_type"])
-        q.set_order("source", "signal", "time_type", "time_value", "geo_type", "geo_value", "issue")
+        def gen_transform(rows):
+            parsed_rows = (parse_row(row, fields_string, fields_int, fields_float) for row in rows)
+            for row in parsed_rows:
+                yield alias_row(row)
+
+    q.set_order("geo_type", "geo_value", "source", "signal", "time_type", "time_value", "issue")
```

Comment:
i think this change of ordering in the db query is causing the slowdown we saw in QA.  the previous order should be able to use an index to essentially get the ordering for free, but this one causes an extra sort step.  can we change it back and then ill run the locust tests again?